README.md

```diff
@@ -0,0 +1,318 @@
+# Creating your own UI with the Smooch iOS SDK
+
+**This guide shows you how to create your own UI for the Smooch iOS SDK.**
+
+The Smooch iOS SDK comes with a rich prebuilt user interface with the option to configure the interface using the Smooch dashboard or the REST API.
+
+If needed you can completely replace Smooch's default UI with your own interface.
+
+Although you can replace the default UI, note that this means rewriting support for all the Smooch message types, that you want to support in your custom UI.
+
+## Overview
+
+The iOS SDK can be initialized without displaying it's default UI. You can then make use of the SDK's messaging APIs to send messages, and it's conversation delegate methods to receive messages.
+
+This guide is separated into two parts:
+- [Part One](#) for setting up a generic messaging UI in iOS
+- [Part Two](#) for dropping Smooch methods in to add the messaging functionality.
+
+It will help to have the [SDK documentation](https://docs.smooch.io/api/ios/index.html) on hand while following this guide.
+
+The complete code for this guide is included in this repository.
+
+## Part One: setting up the UI
+
+In part one we'll set up the UI that we can drop Smooch into in part two.
+
+### 1. Create a new project
+Create a new single view app project ![single view app](images/single_view_app.png) and pick Swift as the language.
+
+### 2. Add UI elements
+In the _Main.storyboard_ file add a Text Field and a Table View. The Text Field will be our message input. The Table View will contain the conversation history.
+
+![text field](images/text_field_element.png)
+
+![table view](images/table_view_element.png)
+
+### 3. Connect UI elements to the view controller
+Using the assistant editor, ctrl drag both UI elements from _Main.storyboard_ into _ViewController.swift_ to connect them to referencing outlets.
+
+Name the Text View _messageInput_  and the Table View _conversationHistory_.
+
+Your _ViewController.swift_ file should now look something like this:
+
+```swift
+class ViewController: UIViewController {
+    @IBOutlet weak var messageInput: UITextField!
+    @IBOutlet weak var conversationHistory: UITableView!
+
+    override func viewDidLoad() {...}
+}
+```
+
+### 4. Handle input from the Text View
+Now we're going to handle the user hitting the enter key while typing so that we can treat their input as a message.
+
+First we'll create a function called _endOfInput_ to handle the event:
+
+```swift
+@objc func endOfInput(){
+    messageInput.resignFirstResponder()
+    let text = messageInput.text!.trimmingCharacters(in: .whitespacesAndNewlines)
+    if text.count > 0 {
+        print("Message: \(text)")
+    }
+    messageInput.text = ""
+}
+```
+
+The _endOfInput_ function is logging the text to the console and resetting the Text View to an empty state. When we implement the Smooch pieces, this is where we'll send a message instead of logging it.
```

Comment:
When we implement the Smooch pieces -> "This is where we'll call into Smooch to send the message in part two